remote/remote_state_s3.go

```diff
@@ -173,20 +212,71 @@ func checkIfVersioningEnabled(s3Client *s3.S3, config *RemoteStateConfigS3, terr
 }
 
 // Create the given S3 bucket and enable versioning for it
-func CreateS3BucketWithVersioning(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
-	if err := CreateS3Bucket(s3Client, config, terragruntOptions); err != nil {
+func CreateS3BucketWithVersioning(s3Client *s3.S3, config *ExtendedRemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
+	if err := CreateS3Bucket(s3Client, &config.remoteStateConfigS3, terragruntOptions); err != nil {
 		return err
 	}
 
-	if err := WaitUntilS3BucketExists(s3Client, config, terragruntOptions); err != nil {
+	if err := WaitUntilS3BucketExists(s3Client, &config.remoteStateConfigS3, terragruntOptions); err != nil {
 		return err
 	}
 
-	if err := EnableVersioningForS3Bucket(s3Client, config, terragruntOptions); err != nil {
+	if err := TagS3Bucket(s3Client, config, terragruntOptions); err != nil {
 		return err
 	}
 
+	if err := EnableVersioningForS3Bucket(s3Client, &config.remoteStateConfigS3, terragruntOptions); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func TagS3Bucket(s3Client *s3.S3, config *ExtendedRemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
+
+	if len(config.S3BucketTags) == 0 {
+		terragruntOptions.Logger.Print("No tags for S3 bucket given.")
+		return nil
+	}
+
+	// There must be one entry in the list
+	var tags = config.S3BucketTags[0]
+	var tagsConverted = convertTags(tags)
+
+	terragruntOptions.Logger.Printf("Tagging S3 bucket with %s", tags)
+
+	putBucketTaggingInput := s3.PutBucketTaggingInput{
+		Bucket: aws.String(config.remoteStateConfigS3.Bucket),
+		Tagging: &s3.Tagging{
+			TagSet: tagsConverted}}
+
+	_, err := s3Client.PutBucketTagging(&putBucketTaggingInput)
+
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
 	return nil
+
+}
+
+func convertTags(tags map[string]string) []*s3.Tag {
+
+	var tagsConverted []*s3.Tag
+
+	var i = 0
```

Comment:
What's `i` for?

Reply:
I used that before I figured out there is an append function ;-) Removed